src/v2/models/AppState.js

```diff
@@ -236,17 +241,22 @@ export default Model.extend({
       && messagesObjs.value.some(messagesObj => messagesObj.i18n?.key.startsWith(keySubStr));
   },
 
-  clearAppStateCache() {
+  clearAppStateCache(doRerender) {
     // clear appState before setting new values
-    this.clear({ silent: true });
+    const attrs = {};
+    for (var key in this.attributes) {
+      if (key !== 'currentFormName' || doRerender) {
+        attrs[key] = void 0;
+      }
+    }
+    this.set(attrs, Object.assign({}, {unset: true, silent: true}));
     // clear cache for derived props.
     this.trigger('cache:clear');
   },
 
   setIonResponse(transformedResponse) {
-    if (!this.shouldReRenderView(transformedResponse)) {
-      return;
-    }
+    const doRerender = this.shouldReRenderView(transformedResponse);
+    this.clearAppStateCache(doRerender);
```

Comment:
this seems to be very cryptic way to re-render the view..  :)

It is much clear to trigger a trigger an event on when to re-render and unbinds re-rendering from changing formName

Reply:
Could be.
Fix the reset currentFormName logic a bit. Plus we do have comment down below to explain the critical part when change currentFormName.